[libpq] Build error

Package: libpq[core,lz4,openssl,zlib]:x64-linux -> 15.3#1

**Host Environment**

- Host: x64-linux
- Compiler: GNU 12.3.1
-    vcpkg-tool version: 2023-03-14-105f7f7e8a5ea10dfac9874ff9d1a6bacba5f454
    vcpkg-scripts version: gpg: Signature made Sat Sep  2 00:34:33 2023 CEST
gpg:                using RSA key 4AEE18F83AFDEB23
gpg: Can't check signature: No public key
4a600e9fe 2023-09-01 (2 days ago)

**To Reproduce**

`vcpkg install `

**Failure logs**

```
-- Using cached postgresql-15.3.tar.bz2.
-- Cleaning sources at /home/jan/.clion-vcpkg/vcpkg/buildtrees/libpq/src/tgresql-15-fbbadfb3b2.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/jan/.clion-vcpkg/vcpkg/downloads/postgresql-15.3.tar.bz2
-- Applying patch unix/fix-configure.patch
-- Applying patch unix/single-linkage.patch
-- Applying patch unix/no-server-tools.patch
-- Applying patch unix/mingw-install.patch
-- Applying patch unix/python.patch
-- Applying patch windows/macro-def.patch
-- Applying patch windows/python_lib.patch
-- Applying patch windows/win_bison_flex.patch
-- Applying patch windows/msbuild.patch
-- Applying patch windows/spin_delay.patch
-- Using source at /home/jan/.clion-vcpkg/vcpkg/buildtrees/libpq/src/tgresql-15-fbbadfb3b2.clean
-- Found external ninja('1.11.1').
-- Getting CMake variables for x64-linux
-- Getting CMake variables for x64-linux-dbg
-- Getting CMake variables for x64-linux-rel
-- Generating configure for x64-linux
-- Finished generating configure for x64-linux
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
-- Installing x64-linux-dbg
-- Building x64-linux-rel
-- Installing x64-linux-rel
CMake Error at scripts/cmake/vcpkg_replace_string.cmake:2 (file):
  file failed to open for reading (No such file or directory):

    /home/jan/.clion-vcpkg/vcpkg/packages/libpq_x64-linux/include/postgresql/server/pg_config.h
Call Stack (most recent call first):
  buildtrees/versioning_/versions/libpq/a62f65ad38aa981b6ff7474e16f739a3ffbc622b/portfile.cmake:141 (vcpkg_replace_string)
  scripts/ports.cmake:147 (include)



```


**Additional context**

<details><summary>vcpkg.json</summary>

```
{
  "name": "postgresql-client",
  "version-string": "1.0.0",
  "builtin-baseline": "c95000e1b5bb62884de08d5e952993c8bced9db6",
  "dependencies": [
    {
      "name": "libpqxx",
      "version>=": "7.7.4"
    }
  ]
}

```
</details>

I have seen #33211 as well, but I am using another triplet (x64-linux). I have also seen the discussion happening in that bug report, but for me it boils down to: What can be done for this to work again? This is a hard blocker here.

--- a/ports/libpq/portfile.cmake
+++ b/ports/libpq/portfile.cmake
@@ -8,6 +8,7 @@ vcpkg_extract_source_archive(
     SOURCE_PATH
     ARCHIVE "${ARCHIVE}"
     PATCHES
+        unix/installdirs.patch
         unix/fix-configure.patch
         unix/single-linkage.patch
         unix/no-server-tools.patch

--- a/ports/libpq/unix/installdirs.patch
+++ b/ports/libpq/unix/installdirs.patch
@@ -1,0 +1,48 @@
+diff --git a/src/Makefile.global.in b/src/Makefile.global.in
+index 5dacc4d..a9a797e 100644
+--- a/src/Makefile.global.in
++++ b/src/Makefile.global.in
+@@ -100,14 +100,14 @@ datarootdir := @datarootdir@
+ 
+ bindir := @bindir@
+ 
+-datadir := @datadir@
++datadir := @datadir@/postgresql
+ ifeq "$(findstring pgsql, $(datadir))" ""
+ ifeq "$(findstring postgres, $(datadir))" ""
+ override datadir := $(datadir)/postgresql
+ endif
+ endif
+ 
+-sysconfdir := @sysconfdir@
++sysconfdir := @sysconfdir@/postgresql
+ ifeq "$(findstring pgsql, $(sysconfdir))" ""
+ ifeq "$(findstring postgres, $(sysconfdir))" ""
+ override sysconfdir := $(sysconfdir)/postgresql
+@@ -116,7 +116,7 @@ endif
+ 
+ libdir := @libdir@
+ 
+-pkglibdir = $(libdir)
++pkglibdir = $(libdir)/postgresql
+ ifeq "$(findstring pgsql, $(pkglibdir))" ""
+ ifeq "$(findstring postgres, $(pkglibdir))" ""
+ override pkglibdir := $(pkglibdir)/postgresql
+@@ -125,7 +125,7 @@ endif
+ 
+ includedir := @includedir@
+ 
+-pkgincludedir = $(includedir)
++pkgincludedir = $(includedir)/postgresql
+ ifeq "$(findstring pgsql, $(pkgincludedir))" ""
+ ifeq "$(findstring postgres, $(pkgincludedir))" ""
+ override pkgincludedir := $(pkgincludedir)/postgresql
+@@ -134,7 +134,7 @@ endif
+ 
+ mandir := @mandir@
+ 
+-docdir := @docdir@
++docdir := @docdir@/postgresql
+ ifeq "$(findstring pgsql, $(docdir))" ""
+ ifeq "$(findstring postgres, $(docdir))" ""
+ override docdir := $(docdir)/postgresql

--- a/ports/libpq/vcpkg.json
+++ b/ports/libpq/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "libpq",
   "version": "15.3",
-  "port-version": 1,
+  "port-version": 2,
   "description": "The official database access API of postgresql",
   "homepage": "https://www.postgresql.org/",
   "license": "PostgreSQL",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4494,7 +4494,7 @@
     },
     "libpq": {
       "baseline": "15.3",
-      "port-version": 1
+      "port-version": 2
     },
     "libpqxx": {
       "baseline": "7.7.4",

--- a/versions/l-/libpq.json
+++ b/versions/l-/libpq.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "859b43f8db33293c411c365d945ecd4ae2cc056b",
+      "version": "15.3",
+      "port-version": 2
+    },
+    {
       "git-tree": "a62f65ad38aa981b6ff7474e16f739a3ffbc622b",
       "version": "15.3",
       "port-version": 1

